### PR TITLE
XTypes: Unaligned CDR

### PR DESCRIPTION
**Pull request by sonndinh**

### CODE REVIEW

File: dds/idl/marshal_generator.cpp [hunk 1]
```diff
@@ -1404,7 +1404,7 @@ namespace {
   }
 
   enum Encoding {
-    encoding_unaligned_cdr,
+    //encoding_unaligned_cdr,
```
Comment (iguessthislldo) [inline dds/idl/marshal_generator.cpp:1407]: I will actually need this enum value for #1846.

File: dds/idl/annotations.h [hunk 2]
```diff
@@ -302,66 +302,66 @@ namespace OpenDDS {
...
     }
 
     void set_all(bool value)
     {
       xcdr1 = value;
       xcdr2 = value;
       xml = value;
-      unaligned_cdr = value;
+      //unaligned_cdr = value;
     }
```
Comment (iguessthislldo) [inline dds/idl/annotations.h:315]: I think all the methods below `set_all` could be actually be removed.

Comment (sonndinh) [reply dds/idl/annotations.h]: I saw some other code calling `not_only_xcdr2` and `not_only_xcdr1`, so I'm keeping them for now.

File: tests/DCPS/DataRepresentation/DataRepresentation.idl [hunk 3]
```diff
@@ -5,7 +5,7 @@ struct DefaultedType {
 
 @topic
 @::OpenDDS::data_representation(XCDR1)
-@::OpenDDS::data_representation(UNALIGNED_CDR)
+//@::OpenDDS::data_representation(UNALIGNED_CDR)
```
[the remark this answers is not in the document]
Comment (sonndinh) [reply tests/DCPS/DataRepresentation/DataRepresentation.idl:8]: No, I can delete it. I just commented it out in the meantime to test CI results.

[the remark this answers is not in the document]
Comment (lij-oci) [reply tests/DCPS/DataRepresentation/DataRepresentation.idl:8]: I reviewed all the changes in this PR and will approve when this comment is resolved.

[the remark this answers is not in the document]
Comment (sonndinh) [reply tests/DCPS/DataRepresentation/DataRepresentation.idl]: It's removed now.

**Review by lij-oci — APPROVED** [review]

**Comment by iguessthislldo** [comment]
Have we talked about the discovery side of these changes yet?

**Comment by sonndinh** [comment]
> Have we talked about the discovery side of these changes yet?

I think we discussed a bit but I haven't totally understood what needs to be done for the discovery side. Could you give a description of what needs to be done on the discovery side?

**Comment by iguessthislldo** [comment]
The last I think we talked about it, the idea was that DataRepresentionQos should be ignored in non-encapsulated situations.

File: dds/idl/marshal_generator.cpp [hunk 4]
```diff
@@ -2162,9 +2162,6 @@ namespace {
     if (repr.xml) {
       values.push_back("DDS::XML_DATA_REPRESENTATION");
     }
-    if (repr.unaligned_cdr) {
-      values.push_back("UNALIGNED_CDR_DATA_REPRESENTATION");
```
Comment (iguessthislldo) [inline dds/idl/marshal_generator.cpp:2166]: If we're not using it anymore, I think `OpenDDS::UNALIGNED_CDR_DATA_REPRESENTATION` can be removed from `DdsDcpsCore.idl`.

Comment (sonndinh) [reply dds/idl/marshal_generator.cpp:2166]: It's removed now.

**Review by mitza-oci — APPROVED** [review]

**Review by iguessthislldo — APPROVED** [review]

**Pull request merged**